src/main.js

```diff
@@ -117,6 +137,19 @@ class PlatformIOVSCodeExtension {
     ));
   }
 
+  terminateUploadTask() {
```

Comment:
Please rewrite this block with `async/await`
```js
  async terminateUploadTask() {
    if(!this._inUploadTask) {
return;
}
      this._inUploadTask = false;
try {
      await vscode.commands.executeCommand("workbench.action.tasks.terminate")
} catch (err) {
console.error(err);
}
        return new Promise((resolve, reject) => {
          setTimeout(() => resolve(), 100);
        });
      });
  }
```

P.S: I didn't test this code. Please valiate.